### ISSUE TITLE
SigSegv after typing wrong namespace

### ISSUE THREAD

**Issue by lekcyjna123** [clangd/clangd#866]
Hi,
i have found that from time to time clangd 12.0.1 crach with SIGSEGV error. Here is minimal example which i was able to get:

```cpp
#include <ranges>
#include <map>
#include <vector>
template<typename T>
  requires  std::ranges::random_access_range<T>
void fit(const T& zbior) 
{
  std::map<int, std::vector<FooBar&>> X;
  int test = std::zbior; // <- line to crach
}
```
When server is started on this file it correctly process it and found errors. There are two: FooBar is undefined and `zbior` isn't in `std`. Additionally there is warning that argument `zbior` isn't used. Now we can try to edit `std::zbior` (in practical use case we probably will fix error in this line by editing it). When we remove one of colons (`:`), exit from editing mode (i use vim) clangd correctly recognize change in this line and inform that `std:` isn't needed. But when we add colon back, then clangd crach.

Summarize of states of "line to crach":
- `int test = std::zbior;` <- OK
- `int test = std:zbior;` <- OK
- `int test = std::zbior;` <- Crach

My observations:
- when instead of `std::ranges::random_access_range<T>` there are hand made concepts there are no problem
- when changed FooBar to int there is no problem
- when FooBar is defined (as `typedef`) in the same file with this example there is no problem
- when FooBar is depended type defined in header file in template class declaration there is problem (this case is part of bigger example which i had reduced)

Practical use case:
I wanted to edit line `int test = zbior` by adding function from `std`. So i started to type on the begining of text "zbior", so there was a moment in which this text was "std::zbior" and whatever i write after that was meaningless because server already crashed.

**Logs**

[clangd_stderrf1xkfacn.log](https://github.com/clangd/clangd/files/7190212/clangd_stderrf1xkfacn.log)

**System information**

Output of `clangd --version`:

clangd version 12.0.1

Editor/LSP plugin:

Vim 8.2 + YouCompleteMe (master from today 18.09.2021)

Operating system:

Arch Linux

**Comment by lekcyjna123** [comment]
Hi,
i retested this issue on snapshot:
```
clangd version 14.0.0 (https://github.com/llvm/llvm-project fa69ccd189694d0b992323738443ac05a2bcb9ca)
Features: linux+grpc
Platform: x86_64-unknown-linux-gnu
```
On this snapshot there is no longer problem.

**Comment by HighCommander4** [comment]
Based on the crash stack, this looks like the issue fixed in #771. The fix should appear in the 13.0 release.
